Server/src/main/java/net/simon987/server/user/UserStatsHelper.java

```diff
@@ -38,7 +38,38 @@ public UserStatsHelper(MongoCollection<Document> users) {
 
         for (Document dbUser : users.find().sort(orderBy).limit(n)) {
             User user = GameServer.INSTANCE.getGameUniverse().getUser((String) dbUser.get("username"));
-            rows.add(new AbstractMap.SimpleEntry<>(user, user.getStats().getInt(statName)));
+            int val = 0;
+            if(user.getStats().getInt(statName) > 0)
+                val = user.getStats().getInt(statName);
+            rows.add(new AbstractMap.SimpleEntry<>(user, val));
+        }
+
+        return rows;
+    }
+
+    /**
+     * Get top n players along with all their stat values, in descending order
+     *
+     * @param n        Maximum number of players
+     * @return Top n players, in User,value format, in descending order
+     */
+    public ArrayList<Map.Entry<User, Map<String, Integer>>> getTopNAll(int n) {
+
+        ArrayList<Map.Entry<User, Map<String, Integer>>> rows = new ArrayList<>();
+
+        ArrayList<Map.Entry<User, ArrayList>> vaults = new ArrayList<>(this.getTopNSetLength("completedVaults", n));
+        ArrayList<Map.Entry<User, Integer>>  deaths = new ArrayList<>(this.getTopN("deathCount", n));
+        ArrayList<Map.Entry<User, Integer>>  time = new ArrayList<>(this.getTopN("executionTime", n));
+        ArrayList<Map.Entry<User, Integer>>  distance = new ArrayList<>(this.getTopN("walkDistance", n));
+        
+        for (int i = 0; i < vaults.size() ; i++) {
+            User user = vaults.get(i).getKey();
+            Map<String, Integer> allStats = new HashMap();
+            allStats.put("completedVaults", vaults.get(i).getValue().size());
+            allStats.put("deathCount", deaths.get(i).getValue());
```

Comment:
This will not work with multiple users because the order of `deaths` and `vaults` is not necessarily the same.
This part will require a more complex database query and you can leave it as it is for now - I'll create the utility method in UserStats when I get the time

Reply:
Okay